### PR TITLE
Incorporate uncertainty

### PR DESCRIPTION
**Pull request by AHsu98**
Added lower level functions to start incorporating uncertainty

### CODE REVIEW

File: src/kreg/uncertainty/lanczos_sample.py
Comment (zhengp0) [inline src/kreg/uncertainty/lanczos_sample.py:1]: Hi @AHsu98! Just want you know that `matfree` package had lots of changes recently. Module `matfree.lanczos` has gone, instead it provides a bunch of other module to accomplish what `funm_vector_product_spd` does. I made a bunch of changes to adapt to the new version, please take a look when you get a chance :)

Comment (AHsu98) [reply src/kreg/uncertainty/lanczos_sample.py:1]: Cool, I’ll take a look tomorrow. Once things calm down I think that we’ll want to experiment with our own samplers and write our own lanczos/chebyshev sampler! I have some ideas for new algorithms for this as well, given the repetitive nature of this.

Comment (kels271828) [reply src/kreg/uncertainty/lanczos_sample.py:1]: Haha, thanks Peng! I ran into issues with this too and had to modify my own version of kreg, but I'm more confident in your updates :)

**Comment by AHsu98** [comment]
Looks good to me! Maybe we should schedule an internal learning on Lanczos algorithm and gaussian sampling to force me to properly learn it enough to write a custom alg for this 🙈

**Comment by zhengp0** [comment]
> Looks good to me! Maybe we should schedule an internal learning on Lanczos algorithm and gaussian sampling to force me to properly learn it enough to write a custom alg for this 🙈

Hahaha, that is a good idea! :)

**Pull request merged**